packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -1,52 +1,93 @@
-  nuxt.hook('vite:extendConfig', async (viteInlineConfig, _env) => {
-    const sourceMapsUploadOptions = moduleOptions.sourceMapsUploadOptions || {};
-
-    if ((sourceMapsUploadOptions.enabled ?? true) && viteInlineConfig.mode !== 'development') {
-      const sentryPlugin = sentryVitePlugin({
-        org: sourceMapsUploadOptions.org ?? process.env.SENTRY_ORG,
-        project: sourceMapsUploadOptions.project ?? process.env.SENTRY_PROJECT,
-        authToken: sourceMapsUploadOptions.authToken ?? process.env.SENTRY_AUTH_TOKEN,
-        telemetry: sourceMapsUploadOptions.telemetry ?? true,
-        sourcemaps: {
-          assets: sourceMapsUploadOptions.sourcemaps?.assets ?? undefined,
-          ignore: sourceMapsUploadOptions.sourcemaps?.ignore ?? undefined,
-          filesToDeleteAfterUpload: sourceMapsUploadOptions.sourcemaps?.filesToDeleteAfterUpload ?? undefined,
-        },
-        _metaOptions: {
-          telemetry: {
-            metaFramework: 'nuxt',
-          },
-        },
-        debug: moduleOptions.debug ?? false,
-      });
+  const sourceMapsUploadOptions = moduleOptions.sourceMapsUploadOptions || {};
+  const sourceMapsEnabled = sourceMapsUploadOptions.enabled ?? true;
 
+  nuxt.hook('vite:extendConfig', async (viteInlineConfig, _env) => {
+    if (sourceMapsEnabled && viteInlineConfig.mode !== 'development') {
+      // Add Sentry plugin
       viteInlineConfig.plugins = viteInlineConfig.plugins || [];
-      viteInlineConfig.plugins.push(sentryPlugin);
-
-      const sourceMapsPreviouslyEnabled = viteInlineConfig.build?.sourcemap;
-
-      if (moduleOptions.debug && !sourceMapsPreviouslyEnabled) {
-        // eslint-disable-next-line no-console
-        console.log('[Sentry]: Enabled source maps generation in the Vite build options.');
-        if (!moduleOptions.sourceMapsUploadOptions?.sourcemaps?.filesToDeleteAfterUpload) {
-          // eslint-disable-next-line no-console
-          console.warn(
-            `[Sentry] We recommend setting the \`sourceMapsUploadOptions.sourcemaps.filesToDeleteAfterUpload\` option to clean up source maps after uploading.
-[Sentry] Otherwise, source maps might be deployed to production, depending on your configuration`,
-          );
-        }
-      }
+      viteInlineConfig.plugins.push(sentryVitePlugin(getPluginOptions(moduleOptions)));
 
+      // Enable source maps
       viteInlineConfig.build = viteInlineConfig.build || {};
       viteInlineConfig.build.sourcemap = true;
+
+      logDebugInfo(moduleOptions, viteInlineConfig.build?.sourcemap);
+    }
+  });
+
+  nuxt.hook('nitro:config', (nitroConfig: NitroConfig) => {
+    if (sourceMapsEnabled && !nitroConfig.dev) {
+      if (nitroConfig.rollupConfig) {
+        // Add Sentry plugin
+        if (!Array.isArray(nitroConfig.rollupConfig.plugins)) {
+          nitroConfig.rollupConfig.plugins = nitroConfig.rollupConfig.plugins ? [nitroConfig.rollupConfig.plugins] : [];
```

Comment:
Would you mind adding a short comment why we are overriding the plugins with themselves but as a single item in an array? :thinking: What is happening here?

Reply:
According to the type it is not always an array and I am converting it to an array to `.push` the plugin later on

Reply:
Question is, is converting it from whatever it is to an array safe?